doc.rkt

```diff
@@ -315,6 +316,44 @@
                                   #:end (Pos #:line line #:char really-indent))
                    #:newText new-text)]))
 
+(define (token-type-encoding token)
+  (index-of *semantic-token-types* (SemanticToken-type token)))
+
+(define (token-modifier-encoding token)
+  (define indexes (indexes-where *semantic-token-modifiers*
+                                 (λ (m) (memq m (SemanticToken-modifiers token)))))
+  (for/sum ([index indexes])
+    (expt 2 index)))
```

Comment:
Why is $2^i$? Do you know where is the spec mention this?

Reply:
It's in [semantic token legend section](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#semanticTokensLegend) :

> Since a token type can have n modifiers, multiple token modifiers can be set by using bit flags, so a tokenModifier value of 3 is first viewed as binary 0b00000011, which means [tokenModifiers[0], tokenModifiers[1]] because bits 0 and 1 are set.

Modifier encoding of a token is a bit flag of its modifiers. Intuitively, the code is

```
(for/fold ([flag 0])
          ([index indexes])
  ; C syntax
  flag | (1 << index)
)
```

`(for/sum ([i indexes]) (expt 2 i))` is a simple equivalent implementation. I should mention this in comment.